### PR TITLE
Add width and height operators

### PR DESCRIPTION
**Pull request by wmww**
More missing geometry operators. More needs to be done (I'm hoping to completely remove displacements, see #1064), but this helps.

### CODE REVIEW

File: include/core/mir/geometry/size.h [hunk 1]
```diff
@@ -65,6 +65,11 @@ inline constexpr Size operator*(Size const& size, Scalar scale)
     return scale*size;
 }
 
+inline constexpr Size operator+(Size const& lhs, Size const& rhs)
+{
+    return Size{lhs.width + rhs.width, lhs.height + rhs.height};
```
Comment (AlanGriffiths) [inline include/core/mir/geometry/size.h]: The other operators introduced here are arguable (while adding widths has a clear semantic, I'm not sure it is useful).

Howeverm _this_ one is wrong. (As discussed in #1064)

Comment (wmww) [reply include/core/mir/geometry/size.h]: Removed

File: include/core/mir/geometry/displacement.h [hunk 2]
```diff
@@ -91,6 +91,11 @@ inline constexpr Displacement operator-(Point const& lhs, Point const& rhs)
     return Displacement{lhs.x - rhs.x, lhs.y - rhs.y};
 }
 
+inline constexpr Displacement operator-(Size const& lhs, Size const& rhs)
+{
+    return Displacement{lhs.width - rhs.width, lhs.height - rhs.height};
+}
```
Comment (AlanGriffiths) [inline include/core/mir/geometry/displacement.h]: Sorry, should have commented on this one too: it is wrong. (As discussed in #1064)

Comment (wmww) [reply include/core/mir/geometry/displacement.h]: Good catch

**Comment by AlanGriffiths** [comment]
The title and commit comment probably need to be tidied up before landing.

And @RAOF deserves a chance to say he doesn't like these operators. ;)

**Comment by wmww** [comment]
bors r=RAOF

**Pull request merged**